pkg/bindings/connection.go

```diff
@@ -150,8 +149,8 @@ func GetConnectionFromContext(ctx context.Context) (*Connection, error) {
 	if c == nil {
 		return nil, errors.New("unable to get connection from context")
 	}
-	conn := c.(Connection)
-	return &conn, nil
+	conn := c.(*Connection)
```

Comment:
Why not just return the pointer?

Reply:
it actually fails to work work like that. it was on the casting side.